From cd9a559a5fa7510474ffeacbfde0d2927f2d50b7 Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Thu, 16 Mar 2023 19:58:49 +0530
Subject: [PATCH 127/281] OpenWrt:
 723-net-phy-aquantia-fix-system-side-protocol-mi.patch

drivers: net: phy: aquantia: fix system side protocol misconfiguration

Do not set up protocols for speeds that are not supported by FW.  Enabling
these protocols leads to link issues on system side.

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>

Change-Id: I3d36483a4cee74fcbcb1ef947a0599cf600a6ba8
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
---
 drivers/net/phy/aquantia_main.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/net/phy/aquantia_main.c b/drivers/net/phy/aquantia_main.c
index 3c67723363ec..38f112dfaf92 100644
--- a/drivers/net/phy/aquantia_main.c
+++ b/drivers/net/phy/aquantia_main.c
@@ -340,10 +340,16 @@ static int aqr_config_aneg_set_prot(struct phy_device *phydev)
 	phy_write_mmd(phydev, MDIO_MMD_VEND1, AQUANTIA_VND1_GSTART_RATE,
 		      aquantia_syscfg[if_type].start_rate);
 
-	for (i = 0; i <= aquantia_syscfg[if_type].cnt; i++)
+	for (i = 0; i <= aquantia_syscfg[if_type].cnt; i++) {
+		u16 reg = phy_read_mmd(phydev, MDIO_MMD_VEND1,
+				       AQUANTIA_VND1_GSYSCFG_BASE + i);
+		if (!reg)
+			continue;
+
 		phy_write_mmd(phydev, MDIO_MMD_VEND1,
 			      AQUANTIA_VND1_GSYSCFG_BASE + i,
 			      aquantia_syscfg[if_type].syscfg);
+	}
 
 	/* wake PHY back up */
 	phy_write_mmd(phydev, MDIO_MMD_VEND1, AQUANTIA_VND1_GLOBAL_SC, 0);
-- 
2.17.1

